cpp/daal/src/data_management/train_test_split.cpp

```diff
@@ -96,6 +96,7 @@ services::Status generateShuffledIndicesImpl(const NumericTablePtr & idxTable, c
     const size_t n         = idxTable->getNumberOfRows();
     const size_t stateSize = rngStateTable->getNumberOfRows();
     // number of generated uints: 1.5x of n for reserve
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, n / 2, 3);
     const size_t nRandomUInts = n * 3 / 2;
```

Comment:
I would add a literal, because these operations can happen in int type
```suggestion
    const size_t nRandomUInts = n * 3lu / 2lu;
```

Reply:
And you need to change the order - as done in the check.

Reply:
Fixed